src/common/url.cpp

```diff
@@ -4,19 +4,33 @@
 
 #include <common/url.h>
 
-#include <event2/http.h>
-
-#include <cstdlib>
 #include <string>
+#include <stdexcept>
+#include <charconv>
 
-std::string urlDecode(const std::string &urlEncoded) {
+std::string UrlDecode(const std::string& url_encoded) {
     std::string res;
-    if (!urlEncoded.empty()) {
-        char *decoded = evhttp_uridecode(urlEncoded.c_str(), false, nullptr);
-        if (decoded) {
-            res = std::string(decoded);
-            free(decoded);
+    res.reserve(url_encoded.size());
+
+    for (size_t i = 0; i < url_encoded.size(); ++i) {
+        char c = url_encoded[i];
+        // Special handling for percent which should be followed by two hex digits
+        // representing an octet values, see RFC 3986, Section 2.1 Percent-Encoding
+        if (c == '%' && i + 2 < url_encoded.size()) {
+            int decoded_value = 0;
+            auto [ptr, ec] = std::from_chars(&url_encoded[i + 1], &url_encoded[i + 3], decoded_value, 16);
```

Comment:
`ptr` seems unused, we might as well inline it to:
```C++
std::from_chars(&url_encoded[i + 1], &url_encoded[i + 3], decoded_value, 16).ec == std::errc{})
```
(note, I've used `std::errc{}` instead, seems we're using that more often in the codebase)

Reply:
I think this is less readable so I will keep it as is but I did take @Sjors 's suggestion to change it to change `ptr` to `_`.